example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExamplePolicyCacheWarmerFactory.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.example.config;
+
+import org.apache.avro.reflect.MapEntry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.example.hrdatagenerator.types.Employee;
+import uk.gov.gchq.palisade.example.util.ExampleFileUtil;
+import uk.gov.gchq.palisade.resource.ParentResource;
+import uk.gov.gchq.palisade.resource.Resource;
+import uk.gov.gchq.palisade.resource.impl.DirectoryResource;
+import uk.gov.gchq.palisade.resource.impl.FileResource;
+import uk.gov.gchq.palisade.resource.impl.SystemResource;
+import uk.gov.gchq.palisade.rule.Rule;
+import uk.gov.gchq.palisade.service.PolicyCacheWarmerFactory;
+import uk.gov.gchq.palisade.service.UserCacheWarmerFactory;
+import uk.gov.gchq.palisade.service.request.Policy;
+
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.stream.StreamSupport;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
+@ConfigurationProperties
+public class ExamplePolicyCacheWarmerFactory implements PolicyCacheWarmerFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExamplePolicyCacheWarmerFactory.class);
+
+    private String type;
+    private String resource;
+    private String owner;
+    private Map<String, String> resourceRules;
+    private Map<String, String> recordRules;
+
+    /**
+     * Constructor with 0 arguments for an example implementation
+     * of the {@link PolicyCacheWarmerFactory} interface
+     */
+    public ExamplePolicyCacheWarmerFactory() {
```

Comment:
Why not have an empty constructor and put these as default values for variable instantiation? That way any other constructor doesn't have to worry about maybe calling `this()`?

Reply:
These are now the default values for the variables. The constructor is empty.